internal/vault/client_factory.go

```diff
@@ -381,6 +425,10 @@ func (m *cachingClientFactory) cacheClient(c Client) (ClientCacheKey, error) {
 		return "", err
 	}
 
+	if c.IsClone() {
+		return "", fmt.Errorf("cannot cache cloned Clients")
+	}
```

Comment:
Why are clone clients rejected here, when it looks like they're added to the cache in `cachingClientFactory.Get()`?

Reply:
Updated in 1a28323f25fc18e3791844589c865f76376b9dde